I have improved the test to not rely on `_ready` and removed the windows hack, the test passes on this branch while fails on current main:
```console
Running Debug|x64 interpreter...
== CPython 3.15.0a1+ (heads/cancel_in_loop:787c9fc5de0, Dec 6 2025, 20:45:00) [MSC v.1944 64 bit (AMD64)]
== Windows-11-10.0.26200-SP0 little-endian
== Python build: debug
== cwd: D:\cpython\build\test_python_worker_24356æ
== CPU count: 12
== encodings: locale=cp1252 FS=utf-8
== resources: all test resources are disabled, use -u option to unskip tests

Using random seed: 1952231759
0:00:00 Run 1 test sequentially in a single process
0:00:00 [1/1] test_asyncio.test_tasks
test_run_coroutine_threadsafe_and_cancel (test.test_asyncio.test_tasks.RunCoroutineThreadsafeTests.test_run_coroutine_threadsafe_and_cancel) ... FAIL

======================================================================
FAIL: test_run_coroutine_threadsafe_and_cancel (test.test_asyncio.test_tasks.RunCoroutineThreadsafeTests.test_run_coroutine_threadsafe_and_cancel)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\cpython\Lib\test\test_asyncio\test_tasks.py", line 3704, in test_run_coroutine_threadsafe_and_cancel
    self.assertTrue(task.cancelled())
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
AssertionError: False is not true

----------------------------------------------------------------------
Ran 1 test in 0.014s

FAILED (failures=1)
Task was destroyed but it is pending!
task: <Task cancelling name='Task-2' coro=<sleep() done, defined at D:\cpython\Lib\asyncio\tasks.py:687> wait_for=<Future cancelled> cb=[_chain_future.<locals>._call_set_state() at D:\cpython\Lib\asyncio\futures.py:397]>
test test_asyncio.test_tasks failed
0:00:00 [1/1/1] test_asyncio.test_tasks failed (1 failure)

== Tests result: FAILURE ==

1 test failed:
    test_asyncio.test_tasks

Total duration: 542 ms
Total tests: run=1 (filtered) failures=1
Total test files: run=1/1 (filtered) failed=1
Result: FAILURE

```

--- a/Misc/NEWS.d/next/Library/2025-11-18-15-48-13.gh-issue-105836.sbUw24.rst
+++ b/Misc/NEWS.d/next/Library/2025-11-18-15-48-13.gh-issue-105836.sbUw24.rst
@@ -1,2 +1,2 @@
 Fix :meth:`asyncio.run_coroutine_threadsafe` leaving underlying cancelled
-asyncio task running
+asyncio task running.

